src/utils/hooks/useYouTube.js

```diff
@@ -0,0 +1,44 @@
+import axios from 'axios';
+
+const API_KEY = 'AIzaSyDE9yryXzFe5HsYFd1uH3FMIOrp6ldrhZM';
+
+export const fetchVideos = (searchValue) => {
+  const APIUrl = `https://www.googleapis.com/youtube/v3/search?key=${API_KEY}&type=video&part=snippet&maxResults=40&q=${searchValue}`;
+  const request = axios.get(APIUrl);
+  return request
+    .then((response) => {
+      if (response.status === 200) {
+        return response.data;
+      }
+      if (response.status === 404) {
+        throw new Error('Not found');
+      }
+      if (!response.ok) {
+        throw new Error('Sorry, something went wrong');
+      }
+    })
+    .catch((error) => {
+      console.log(error);
+    });
+};
+
+export const fetchVideoById = (id) => {
+  const API_URL = `https://www.googleapis.com/youtube/v3/videos?part=snippet&id=${id}&key=${API_KEY}`;
+  const request = axios.get(API_URL);
+  return request
+    .then((response) => {
+      if (response.status === 200) {
+        console.log(response);
+        return response.data;
+      }
+      if (response.status === 404) {
+        throw new Error('Not found');
+      }
+      if (!response.ok) {
+        throw new Error('Sorry, something went wrong');
+      }
+    })
+    .catch((error) => {
+      console.log(error.message);
+    });
+};
```

Comment:
Very nice api calls logic separation. 

Just a note: this functions are not react hooks, are JS vanilla functions.

BUT it's ok to have this functions defined, is a good practice to have this kind of logic separated from react components.

Probably you could put those functions on another folder like `services`, because they are actually a service api call logic things.

And then, you can create custom hook and use those functions. 

Remember that the hooks starts with "use" word. 

The purpose of the react hooks is abstract and reuse logic from components.

Think how can you use a custom hook in order to fetch the api calls.